source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -754,6 +770,27 @@ void Filter::onMaxDownstreamConnectionDuration() {
   read_callbacks_->connection().close(Network::ConnectionCloseType::NoFlush);
 }
 
+void Filter::onAccessLogFlushInterval() {
+  for (const auto& access_log : config_->accessLogs()) {
+    access_log->log(nullptr, nullptr, nullptr, getStreamInfo());
```

Comment:
Is there any way to distinguish intermediate log from the final one when the session is closed? I see in the comments above to use uuid for correlation, so multiple intermediate updates are linked and do not look as separate sessions, but there should be a final log which must inform that the session is completed. Otherwise it will look as another intermediate update and mess the stats.

Reply:
Yeah, of course. The intermediate log entries have specified flag `intermediate_log_entry = true`.

Reply:
Ah, yes - you are right. I see it for grpc sink. But I am guessing that entries flushed to a file/stdout will be undistinguished, correct?

Reply:
> Ah, yes - you are right. I see it for grpc sink. But I am guessing that entries flushed to a file/stdout will be undistinguished, correct?

Yes, for now. It would be easy to add a new command to support distinguishing in the file/stdout in the future.

Reply:
I think we should not base or infer a concrete state or behaviour based on the presence or not of another field(Duration in this case) as it could potentially change in the feature and the connection/dependency created to infer the state(intermediate or final) can break.

Also, I think that having %DURATION% on intermediate `access_log` is quite valuable for the consumer of the entry.
For instance, if the consumer would like to know how many bytes per minute have been send, knowing the duration is crucial and make processing way easier, otherwise it would need to know when the last intermediate entry was send to calculate the duration(and it will not be exact).

Reply:
A follow up question based on my previous message:

On each intermediate message send, what will the values in the different properties be? Concretely metric style properties like for instance byes received/send found at `connection_properties { received_bytes: 6 sent_bytes: 33 }` or perhaps even duration?
Would they be an ever increasing value, or will they be reset and only send the observed values during the `access_log_flush_interval`?

Reply:
The `received/send` would be ever increasing values from the connection was established.